cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/StructuredTable.java

```diff
@@ -34,17 +34,19 @@
    * The fields contain both the primary key and the rest of the columns to write.
    *
    * @param fields the fields to write
+   * @throws InvalidFieldException if any of the fields are not part of the table schema, or the types do not match.
    */
-  void write(Collection<Field<?>> fields);
+  void write(Collection<Field<?>> fields) throws InvalidFieldException;
```

Comment:
I was more wondering about IOException or similar - the underlying storage implementation would throw that, right?

Reply:
As of now the storage implementation exceptions get thrown during commit as TransactionFailureException. We can keep the current behavior, unless the SQL  implementation needs a change

Reply:
Added a TODO for this